src/CommonLib/utility.cxx

```diff
@@ -627,3 +627,44 @@ UTILITY::LogEvent(
         fwprintf(stderr, L"ERROR: %s\n", pstrMsg);
     }
 }
+
+HRESULT
+UTILITY::GetExpandedEnvironmentString(
+    _In_    PCWSTR pcwzString,
+    _Inout_ STRU* struExpandedString
+)
+{
+    HRESULT hr = S_OK;
+    DWORD dwNumBytesRead = ExpandEnvironmentStringsW(pcwzString,
+        struExpandedString->QueryStr(),
+        struExpandedString->QuerySizeCCH());
+
+    if (dwNumBytesRead == 0)
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        goto Finished;
+    }
+    else if (dwNumBytesRead >= struExpandedString->QuerySizeCCH())
+    {
+        // If the buffer is too small, resize and read again.
+        if (FAILED(hr = struExpandedString->Resize(dwNumBytesRead + 1)))
```

Comment:
Are you adding a single byte?

Reply:
This is for the null terminator.

Reply:
Add a test for resizing scenario

Reply:
Done. Do you want the E2E scenario with a custom web.config here?

Reply:
I would prefer having functional tests that does it if it doesn't take too long.

Reply:
Added two functional tests. They are a bit verbose. I would like to create a follow up PR to make a helper that creates/deploys the application and returns the deployment result.